updater/spec/dependabot/updater_spec.rb

```diff
@@ -10,77 +10,97 @@
 require "dependabot/service"
 
 RSpec.describe Dependabot::Updater do
-  subject(:updater) do
-    Dependabot::Updater.new(
-      service: service,
-      job_id: 1,
-      job: job,
-      dependency_files: dependency_files,
-      base_commit_sha: "sha",
-      repo_contents_path: repo_contents_path
-    )
+  # rubocop:disable Metrics/MethodLength
+  def build_job(requested_dependencies: nil, allowed_updates: default_allowed_updates,
+                existing_pull_requests: [], ignore_conditions: [], security_advisories: [],
+                experiments: {}, updating_a_pull_request: false, security_updates_only: false)
+    @build_job ||=
+      Dependabot::Job.new(
+        token: "token",
+        dependencies: requested_dependencies,
+        allowed_updates: allowed_updates,
+        existing_pull_requests: existing_pull_requests,
+        ignore_conditions: ignore_conditions,
+        security_advisories: security_advisories,
+        package_manager: "bundler",
+        source: {
+          "provider" => "github",
+          "repo" => "dependabot-fixtures/dependabot-test-ruby-package",
+          "directory" => "/",
+          "branch" => nil,
+          "api-endpoint" => "https://api.github.com/",
+          "hostname" => "github.com"
+        },
+        credentials: [
+          {
+            "type" => "git_source",
+            "host" => "github.com",
+            "username" => "x-access-token",
+            "password" => "github-token"
+          },
+          {
+            "type" => "random",
+            "secret" => "codes"
+          }
+        ],
+        lockfile_only: false,
+        requirements_update_strategy: nil,
+        update_subdependencies: false,
+        updating_a_pull_request: updating_a_pull_request,
+        vendor_dependencies: false,
+        experiments: experiments,
+        commit_message_options: {
+          "prefix" => "[bump]",
+          "prefix-development" => "[bump-dev]",
+          "include-scope" => true
+        },
+        security_updates_only: security_updates_only
+      )
+  end
+  # rubocop:enable Metrics/MethodLength
+
+  def build_updater(service: build_service, job: build_job, dependency_files: default_dependency_files)
+    @build_updater ||=
```

Comment:
Same issue with memoizing the result.

Reply:
Addressed in [9f50771](https://github.com/dependabot/dependabot-core/pull/6729/commits/9f50771aaaf4e3bf8478d5b6dc199359b60c3284)